TN-551 extracting inline JS from template files

achen2401 opened the pull request:
extracting inline js from profile template files
general template file clean-ups
- create dedicated template file for staff and patient
- bug fixes

pieces of code remain the same, just been moved around with slight modifications 
I think there are still some in-line JS in the code, but will address those in a different PR

**not for this coming release**

pep8speaks commented:
Hello @achen2401! Thanks for submitting the PR.







 - In the file [`portal/views/staff.py`](https://github.com/uwcirg/true_nth_usa_portal/blob/24f47dfb936c4a4932d4feaa7e20ba272e213566/portal/views/staff.py), following are the PEP8 issues :

> [Line 70:1](https://github.com/uwcirg/true_nth_usa_portal/blob/24f47dfb936c4a4932d4feaa7e20ba272e213566/portal/views/staff.py#L70): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 83:5](https://github.com/uwcirg/true_nth_usa_portal/blob/24f47dfb936c4a4932d4feaa7e20ba272e213566/portal/views/staff.py#L83): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 84:5](https://github.com/uwcirg/true_nth_usa_portal/blob/24f47dfb936c4a4932d4feaa7e20ba272e213566/portal/views/staff.py#L84): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '

pbugni commented on portal/templates/profile/profile_macros.html:
```diff
@@ -0,0 +1,1009 @@
+            <div class="profile-radio-list">
+                <div class="checkbox">
+                    <label>
+                        <input type="checkbox" name="race" value="1002-5">{{ _("American Indian or Alaska Native") }}
```
In the database, yes.  I'd call that a separate issue, but yes, this could (and probably should) be a dynamic lookup)

pbugni commented on portal/templates/profile/profile_macros.html:
```diff
@@ -0,0 +1,1009 @@
+    <div id="profileTimeZoneGroup" class="form-group timezone-container flex-item">
+        <label>Time Zone</label>
+        <select class="form-control" id="profileTimeZone" data-save-container-id="profileTimeZoneGroup">
+            <option value="UTC">-- Select --</option>
```
I could make that possible, but at the moment, we take a given TZ and confirm it's legit via:

https://github.com/uwcirg/true_nth_usa_portal/blob/develop/portal/models/extension.py#L85-L87
```
            # Confirm it's a recognized timezone
            try:
                pytz.timezone(timezone)
            except pytz.exceptions.UnknownTimeZoneError:
                abort(400, "Unknown Timezone: '{}'".format(timezone))
```

pbugni commented on portal/views/staff.py:
```diff
@@ -64,6 +68,35 @@ def staff_profile_create():
+    # org list should include all orgs under the current user's org(s)
+    OT = OrgTree()
+    org_list = set()
+    for org in current_user().organizations:
```
Hmm, `OrgTree` has many methods, but I don't see that in particular.  A user has a set of organizations at `user.organizations`, but it's at the arbitrary level of leaf or any other level depending on product business rules.  Happy to add, just tell me what you mean by "collated".

ivan-c replied:
Sorry, that should probably be a User model method (not Org)

achen2401 commented on portal/static/js/main.js:
```diff
@@ -4799,163 +6011,8 @@ var FieldLoaderHelper = function () {
-      return selectOptions;
-};
-//Promise polyfill - IE doesn't support Promise - so need this
-!function(e){function n(){}function t(e,n){return function(){e.apply(n,arguments)}}function o(e){if("object"!=typeof this)throw new TypeError("Promises must be constructed via new");if("function"!=typeof e)throw new TypeError("not a function");this._state=0,this._handled=!1,this._value=void 0,this._deferreds=[],s(e,this)}function i(e,n){for(;3===e._state;)e=e._value;return 0===e._state?void e._de ...
```
@ivan, I switched to using lib from cdn (see my changes) as you have suggested previously.

achen2401 commented on portal/static/js/main.js:
```diff
@@ -1883,132 +1966,1349 @@ var assembleContent = {
+    this.extension = [];
+};
+
+var OrgTool = function() {
```
@ivan-c  Ivan, I will, right now, the file has quite a number of dependencies on other variables declared in main.js.  I need to figure out how to resolve/refactor that.  I think I will do that in a different PR.

achen2401 commented:
@ivan-c Ivan, I think I addressed all of your comments.  Thank you very much again for looking over my changes.  Let me know if you see any more issues with my second round of fixes.

ivan-c commented on portal/eproms/templates/eproms/base.html:
```diff
@@ -98,6 +98,11 @@ <h2 class="modal-title">{{ _("TrueNTH") }}</h2>
 <script src="https://unpkg.com/i18next-browser-languagedetector@2.0.0/i18nextBrowserLanguageDetector.min.js"></script>
 <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-datepicker/1.6.1/js/bootstrap-datepicker.min.js" async></script>
 <script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-table/1.11.1/bootstrap-table.min.js"></script>
+<script src="{{ url_for('static', filename='js/validator.js') }}"></script>
```
Can you use this [CDNJS-hosted version](https://cdnjs.com/libraries/1000hz-bootstrap-validator)?

achen2401 commented:
@ivan-c Ivan, I think I addressed all the issues you raised.  Let me know if you see any issue with my latest fixes.
